yt_dlp/extractor/twitch.py

```diff
@@ -525,6 +528,82 @@ def _extract_storyboard(self, item_id, storyboard_json_url, duration):
                 } for path in images],
             }
 
+    def _extract_chat(self, vod_id):
+        chat_history = []
+        has_more_pages = True
+        retry_sleep = 5
+        max_retries = 3
+        retries = 0
```

Comment:
Use `self.RetryManager`. It has sleep and retry count built in according to user input.

Reply:
Simple example: https://github.com/yt-dlp/yt-dlp/blob/bdd0b75e3f41ff35440eda6d395008beef19ef2f/yt_dlp/extractor/agora.py#L234-L239

Reply:
how do I set the amount to sleep? if I just don't do anything with it, then it retries 3 times without a delay.
I see `sleep_func=self.get_param('retry_sleep_functions', {}).get('extractor')` in the error callback, but I don't understand how/where to set that (the `get_param` currently defaults to the empty dict)

Reply:
User sets it with `--retry-sleep extractor:5` etc. You don't need to handle the sleep. The retrymanager already does.

Reply:
Oh, I see. It was just unusual that it defaults to no sleep